### ISSUE TITLE
rvc -l hangs after disconnecting using rvc -d all

### ISSUE THREAD

**Issue by erikbor** [riboseinc/rvc#17]
I start rvcd with a valid json configuration file:

```sh
$ cat /Users/test/.setup/vpn/vpn.json
[
  {
    "name": "stagingfoobar.baz",
    "ovpn": "/Users/test/test.ovpn",
    "autoconnect": "true",
    "up-script": "",
    "down-script": ""
  }
]
$ sudo ./src/rvcd -c /Users/test/.setup/vpn/vpn.json
Fri 18 Aug 2017 12:08:06        INFO : Main: Initializing rvcd context(at rvcd.c:163)
Fri 18 Aug 2017 12:08:06        INFO : CMD: Initializing rvcd command processor(at cmd.c:496)
Fri 18 Aug 2017 12:08:06        INFO : VPN: Initializing VPN connection manager(at vpn.c:1148)
Fri 18 Aug 2017 12:08:06        INFO : CMD: Starting rvcd command processing thread(at cmd.c:400)
Fri 18 Aug 2017 12:08:06        INFO : VPN: Reading configuration from '/Users/test/.setup/vpn/vpn.json'(at vpn.c:884)
Fri 18 Aug 2017 12:08:06        INFO : VPN: Added VPN configuration with name 'stagingfoobar.baz', openvpn path '/Users/test/test.ovpn'(at vpn.c:86)
Fri 18 Aug 2017 12:08:06        INFO : VPN: Starting VPN connection monitoring thread with name 'stagingfoobar.baz'(at vpn.c:1059)
```

So far so good, in another terminal I will execute rvc with '-l':

```sh
$ ./rvc -l
name: stagingfoobar.baz
		profile: /Users/test/test.ovpn
		auto-connect: Disabled
		up-script: none
		down-script:none
```

Now in the server terminal I see:
```
Fri 18 Aug 2017 12:08:13        INFO : CMD: Received command '{ "cmd": "list", "json": false }'(at cmd.c:314)
Fri 18 Aug 2017 12:08:13        INFO : CMD: Processing 'list' command(at cmd.c:172)
Fri 18 Aug 2017 12:08:13        INFO : VPN: Writing VPN connection list into buffer(at vpn.c:1012)
Fri 18 Aug 2017 12:08:13        INFO : CMD: Sending response(at cmd.c:116)
Fri 18 Aug 2017 12:08:13        INFO : CMD: Response string is 
'name: stagingfoobar.baz
                profile: /Users/test/test.ovpn
                auto-connect: Disabled
                up-script: none
                down-script:none
'
(code: 0)(at cmd.c:151)
```
Which is also good.

Now I send a disconnect for all (even though there is no active connection):
```sh
$ ./rvc -d all -j
{ "code": 0 }
```

The server shows:
```
Fri 18 Aug 2017 12:12:05        INFO : VPN: Sending SIGTERM signal to OpenVPN process with connection name 'stagingfoobar.baz'(at vpn.c:239)
Fri 18 Aug 2017 12:12:05        WARN : VPN: Couldn't send 'signal SIGTERM
' command to OpenVPN management socket(err:38)(at vpn.c:141)
Fri 18 Aug 2017 12:12:05        WARN : VPN: OpenVPN process isn't responding. Try killing it in force...(at vpn.c:293)
Fri 18 Aug 2017 12:12:05        INFO : VPN: Stopping VPN connection with name 'stagingfoobar.baz' has succeeded(at vpn.c:298)
Fri 18 Aug 2017 12:12:05        INFO : Main: Received signal 15(at rvcd.c:60)
Fri 18 Aug 2017 12:12:05        INFO : Main: Finalizing rvcd context(at rvcd.c:187)
```

After this rvc just hangs when doing a '-l':
```sh
$ ./rvc -l

```

**Comment by erikbor** [comment]
And the rvcd also hangs btw (even CTRL-C does not help):

```
Fri 18 Aug 2017 12:12:05        INFO : Main: Received signal 15(at rvcd.c:60)
Fri 18 Aug 2017 12:12:05        INFO : Main: Finalizing rvcd context(at rvcd.c:187)
^C^C^C^C^C^C^C^C^C^C^C
```

**Comment by jjr840430** [comment]
Fixed on #31
